fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/LogicalResultSinkToShortCircuitPointQuery.java

```diff
@@ -0,0 +1,67 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.EqualTo;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.qe.ConnectContext;
+
+import com.google.common.collect.Sets;
+
+import java.util.Set;
+
+/**
+ * short circuit query optimization
+ * pattern : select xxx from tbl where key = ?
+ */
+public class LogicalResultSinkToShortCircuitPointQuery extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalResultSink(logicalProject(logicalFilter(logicalOlapScan())))
+                .when(r -> r.child().child().child().getTable().getEnableLightSchemaChange())
+                .when(r -> r.child().child().child().getTable().getEnableUniqueKeyMergeOnWrite())
+                .when(r -> r.child().child().child().getTable().storeRowColumn())
```

Comment:
u can add when after logicalProject, logicalFilter or logicalOlapScan to avoid call too many times `child()`

Reply:
done